src/Features/CSharp/Portable/AddAccessibilityModifiers/CSharpAddAccessibilityModifiersDiagnosticAnalyzer.cs

```diff
@@ -78,16 +78,54 @@ public CSharpAddAccessibilityModifiersDiagnosticAnalyzer()
 
             // If they already have accessibility, no need to report anything.
             var accessibility = generator.GetAccessibility(member);
-            if (accessibility != Accessibility.NotApplicable)
+
+            var isOmit = option.Value == AccessibilityModifiersRequired.OmitIfDefault;
+
+            if (isOmit)
             {
-                return;
+                if (accessibility == Accessibility.NotApplicable)
+                {
+                    return;
+                }
+
+                // Check for default modifiers in namespace and outside of namespace
+                var parentKind = member.Parent.Kind();
+                if (parentKind == SyntaxKind.CompilationUnit ||
+                    parentKind == SyntaxKind.NamespaceDeclaration)
+                {
+                    // Default is internal
+                    if (accessibility != Accessibility.Internal)
+                    {
+                        return;
+                    }
+                }
+
+                if (parentKind == SyntaxKind.ClassDeclaration ||
+                    parentKind == SyntaxKind.StructDeclaration)
+                {
+                    // Inside a type, default is private
+                    if (accessibility != Accessibility.Private)
+                    {
+                        return;
+                    }
+                }
+            }
+            else
+            {
+                if (accessibility != Accessibility.NotApplicable)
+                {
+                    return;
+                }
             }
 
             // Missing accessibility.  Report issue to user.
             var additionalLocations = ImmutableArray.Create(member.GetLocation());
+            var properties = ImmutableDictionary<string, string>.Empty.Add(
```

Comment:
:bulb: Use a static readonly field for each of the potential property maps.

💭 Really, this dictionary could probably be omitted altogether, since this diagnostic always means "toggle the current state to fix".